Internal/Debug App/Sources/View Controllers/MerchantHeadlessViewController.swift

```diff
@@ -0,0 +1,274 @@
+//
+//  MerchantHeadlessViewController.swift
+//  Debug App
+//
+//  Created by Boris on 20.6.23..
+//  Copyright © 2023 Primer API Ltd. All rights reserved.
+//
+
+import UIKit
+import PrimerSDK
+
+class MerchantHeadlessViewController: UIViewController, PrimerHeadlessUniversalCheckoutDelegate {
+    
+    var settings: PrimerSettings!
+    var clientSession: ClientSessionRequestBody?
+    var clientToken: String?
+
+    var redirectManager: PrimerHeadlessUniversalCheckout.NativeUIManager?
+    var logs: [String] = []
+    var primerError: Error?
+    var checkoutData: PrimerCheckoutData?
+
+    private var paymentId: String?
+    private var activityIndicator: UIActivityIndicatorView?
+
+    class func instantiate(settings: PrimerSettings, clientSession: ClientSessionRequestBody?, clientToken: String?) -> MerchantHeadlessViewController {
+        let mcvc = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "MerchantHeadlessViewController") as! MerchantHeadlessViewController
+        mcvc.settings = settings
+        mcvc.clientSession = clientSession
+        mcvc.clientToken = clientToken
+        return mcvc
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        PrimerHeadlessUniversalCheckout.current.delegate = self
+        PrimerHeadlessUniversalCheckout.current.uiDelegate = self
+    }
+    
+    // MARK: - HELPERS
+    
+    private func showLoadingOverlay() {
```

Comment:
(Maybe not for now, but) it would be nice if those are moved as `UIViewController` extensions so we can reuse them among view controllers